// @libphp/almalinux-9-v85/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRoot = getRoot;
exports.getComposer = getComposer;
exports.getPhp = getPhp;
exports.getPhpIni = getPhpIni;
exports.getPhpFpm = getPhpFpm;
exports.getPhpFpmIni = getPhpFpmIni;
exports.getPhpCgi = getPhpCgi;
exports.getPhpFiles = getPhpFiles;
exports.getPhpModulesFiles = getPhpModulesFiles;
exports.getSharedLibsFiles = getSharedLibsFiles;
exports.getFiles = getFiles;
const path_1 = __importDefault(require("path"));
const fs_1 = __importDefault(require("fs"));
const NATIVE_DIR = path_1.default.resolve(__dirname, "../native");
function getRoot() {
    return path_1.default.dirname(__filename);
}
function getComposer() {
    return path_1.default.resolve(NATIVE_DIR, "php/composer");
}
function getPhp() {
    return path_1.default.resolve(NATIVE_DIR, "php/php");
}
function getPhpIni() {
    return path_1.default.resolve(NATIVE_DIR, "php/php.ini");
}
function getPhpFpm() {
    return path_1.default.resolve(NATIVE_DIR, "php/php-fpm");
}
function getPhpFpmIni() {
    return path_1.default.resolve(NATIVE_DIR, "php/php-fpm.ini");
}
function getPhpCgi() {
    return path_1.default.resolve(NATIVE_DIR, "php/php-cgi");
}
function getPhpFiles() {
    return walk(path_1.default.resolve(NATIVE_DIR, "php"), { prefix: "php" });
}
function getPhpModulesFiles() {
    return walk(path_1.default.resolve(NATIVE_DIR, "php/modules"), { prefix: "php/modules" });
}
async function getSharedLibsFiles() {
    return walk(path_1.default.resolve(NATIVE_DIR, "lib"), { prefix: "lib" });
}
async function getFiles() {
    return {
        ...await getPhpFiles(),
        ...await getPhpModulesFiles(),
        ...await getSharedLibsFiles(),
    };
}
async function walk(folder, options) {
    const walker = await fs_1.default.promises.readdir(folder, { withFileTypes: true });
    const files = walker.filter(item => !item.isDirectory());
    return Object.assign({}, ...files.map(file => {
        const filename = (options === null || options === void 0 ? void 0 : options.prefix) ? path_1.default.join(options.prefix, file.name) : file.name;
        const filepath = path_1.default.join(folder, file.name);
        return { [filename]: filepath };
    }));
}
